CTOR 1176: Enhance FreeBSD storage - Add default ZFS support (#3821)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/operatingsystems-freebsd-snmp.md

| Service Alias | Service Template              | Service Description                                                                                                                                |
|:--------------|:------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------|

> | Cpu           | OS-FreeBSD-Cpu-SNMP-custom    | Check the rate of utilization of CPU for the machine. This check can give the average CPU utilization rate and the rate per CPU for multi-core CPU |

Reviewer: ```suggestion
| Cpu           | OS-FreeBSD-Cpu-SNMP-custom    | Check the rate of CPU utilization for the machine. This check can give the average CPU utilization rate and the rate per CPU for multi-core CPU |
```

| Load          | OS-FreeBSD-Load-SNMP-custom   | Check the server load average                                                                                                                      |

> | Memory        | OS-FreeBSD-Memory-SNMP-custom | Check the rate of the utilization of memory                                                                                                        |

Reviewer: ```suggestion
| Memory        | OS-FreeBSD-Memory-SNMP-custom | Check the rate memory utilization                                                                                                        |
```

> | Memory        | OS-FreeBSD-Memory-SNMP-custom | Check the memory usage (RAM)                                                                                                                       |

Reviewer: ```suggestion
| Memory        | OS-FreeBSD-Memory-SNMP-custom | Check memory usage (RAM)                                                                                                                       |
```

> | Swap          | OS-FreeBSD-Swap-SNMP-custom   | Check the virtual memory usage                                                                                                                     |

Reviewer: ```suggestion
| Swap          | OS-FreeBSD-Swap-SNMP-custom   | Check virtual memory usage                                                                                                                     |
```

> The services listed above are created automatically when the **OS-FreeBSD-SNMP-custom** host template is used.

</TabItem>
<TabItem value="Not attached to a host template" label="Not attached to a host template">

| Service Alias        | Service Template                            | Service Description                                                                                                    | Discovery  |
|:---------------------|:--------------------------------------------|:-----------------------------------------------------------------------------------------------------------------------|:----------:|

> | Disk-Generic-Id      | OS-FreeBSD-Disk-Generic-Id-SNMP-custom      | Check the rate of free space on the disk. For each checks the name of the disk will appear                             |            |

Reviewer: ```suggestion
| Disk-Generic-Id      | OS-FreeBSD-Disk-Generic-Id-SNMP-custom      | Check the rate of free space on the disk. For each check the name of the disk will appear                             |            |
```

> | Disk-Generic-Name    | OS-FreeBSD-Disk-Generic-Name-SNMP-custom    | Check the rate of free space on the disk. For each checks the mount pont of the disk will appear                       |            |

Reviewer: ```suggestion
| Disk-Generic-Name    | OS-FreeBSD-Disk-Generic-Name-SNMP-custom    | Check the rate of free space on the disk. For each check the mount point of the disk will appear                       |            |
```

> | Disk-Global          | OS-FreeBSD-Disk-Global-SNMP-custom          | Check the rate of free space on disks. For each checks the mount point of disks will appear                            | X          |

Reviewer: ```suggestion
| Disk-Global          | OS-FreeBSD-Disk-Global-SNMP-custom          | Check the rate of free space on disks. For each check the mount point of disks will appear                            | X          |
```

> | Disk-IO              | OS-FreeBSD-Disk-IO-SNMP-custom              | Check access disk of the disk. For each check the name of the disk will appear "label" rather than the letter assigned |            |

Reviewer: ```suggestion
| Disk-IO              | OS-FreeBSD-Disk-IO-SNMP-custom              | Check disk access. For each check the name of the disk will appear as a "label" rather than the letter assigned |            |
```

#### Host discovery

| Rule name       | Description                                                                                                                                                                                                                                     |
|:----------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| SNMP Agents     | Discover your resources through an SNMP subnet scan. You need to install the [Generic SNMP](./applications-protocol-snmp.md) connector to get the discovery rule and create a template mapper for the **OS-FreeBSD-SNMP-custom** host template. |

More information about discovering hosts automatically is available on the [dedicated page](/docs/monitoring/discovery/hosts-discovery).

#### Service discovery

| Rule name                    | Description                                                   |
|:-----------------------------|:--------------------------------------------------------------|

> | OS-FreeBSD-SNMP-Disk-Name    | Discover the disk partitions and monitor space occupation     |

Reviewer: ```suggestion
| OS-FreeBSD-SNMP-Disk-Name    | Use the disk name to discover the disk partitions and monitor space occupation     |
```

> | OS-FreeBSD-SNMP-Disk-Path    | Discover the disk partitions and monitor space occupation     |

Reviewer: ```suggestion
| OS-FreeBSD-SNMP-Disk-Path    | Use the disk path to discover the disk partitions and monitor space occupation    |
```

</TabItem>
<TabItem value="Disk-*" label="Disk-*">

> Coming soon

Reviewer: coming soon

| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options).                                                                                              | --filter-perfdata='storage.space\|used\|free' |             |

</TabItem>
<TabItem value="Disk-Generic-Name" label="Disk-Generic-Name">

| Macro        | Description                                                                                                                                                                                     | Default value                                 | Mandatory   |
|:-------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:----------------------------------------------|:-----------:|

> | DISKNAME     | Set the storage (number expected) example: 1, 2,... (empty means 'check all storage')                                                                                                           |                                               |             |

Reviewer: ```suggestion
| DISKNAME     | Set the storage name (empty means 'check all storage')                                                                                                           |                                               |             |
```

> | TRANSFORMSRC | Modify the storage name displayed by using a regular expression.  Example: adding --display-transform-src='dev' --display-transform-dst='run'  will replace all occurrences of 'dev' with 'run' |                                               |             |

Reviewer: ```suggestion
| TRANSFORMSRC | Modify the storage name displayed by using a regular expression.  Example: adding --display-transform-src='dev' --display-transform-dst='run' will replace all occurrences of 'dev' with 'run' |                                               |             |
```

> | TRANSFORMDST | Modify the storage name displayed by using a regular expression.  Example: adding --display-transform-src='dev' --display-transform-dst='run'  will replace all occurrences of 'dev' with 'run' |                                               |             |

Reviewer: ```suggestion
| TRANSFORMDST | Modify the storage name displayed by using a regular expression.  Example: adding --display-transform-src='dev' --display-transform-dst='run' will replace all occurrences of 'dev' with 'run' |                                               |             |
```

| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options).                                                                                              | --filter-perfdata='storage.space\|used\|free' |             |

</TabItem>
<TabItem value="Disk-Global" label="Disk-Global">

| Macro        | Description                                                                                                                                                                                     | Default value                                           | Mandatory   |
|:-------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:--------------------------------------------------------|:-----------:|

> | FILTER       | Set the storage (number expected) example: 1, 2,... (empty means 'check all storage')                                                                                                           | .*                                                      |             |

Reviewer: ```suggestion
| FILTER       | Set the storage name, it can be a regex (empty means 'check all storage')                                                                                                           | .*                                                      |             |
```

> | TRANSFORMSRC | Modify the storage name displayed by using a regular expression.  Example: adding --display-transform-src='dev' --display-transform-dst='run'  will replace all occurrences of 'dev' with 'run' |                                                         |             |

Reviewer: ```suggestion
| TRANSFORMSRC | Modify the storage name displayed by using a regular expression.  Example: adding --display-transform-src='dev' --display-transform-dst='run' will replace all occurrences of 'dev' with 'run' |                                                         |             |
```

> | TRANSFORMDST | Modify the storage name displayed by using a regular expression.  Example: adding --display-transform-src='dev' --display-transform-dst='run'  will replace all occurrences of 'dev' with 'run' |                                                         |             |

Reviewer: ```suggestion
| TRANSFORMDST | Modify the storage name displayed by using a regular expression.  Example: adding --display-transform-src='dev' --display-transform-dst='run' will replace all occurrences of 'dev' with 'run' |                                                         |             |
```

| CRITICAL     | Critical threshold                                                                                                                                                                              | 90                                                      |             |
| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options).                                                                                              | --verbose --filter-perfdata='storage.space\|used\|free' |             |

</TabItem>
<TabItem value="Disk-IO" label="Disk-IO">

| Macro         | Description                                                                                        | Default value     | Mandatory   |
|:--------------|:---------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | DISKNAME      | Set the device (number expected) example: 1, 2,... (empty means 'check all devices')               |                   |             |

Reviewer: ```suggestion
| DISKNAME      | Set the storage name, it can be a regex (empty means 'check all storage')             |                   |             |
```

|:-------------|:---------------------------------------------------------------------------------------------------|:------------------|:-----------:|
| CRITICAL     | Threshold                                                                                          | 30                |             |
| WARNING      | Threshold                                                                                          | 10                |             |
| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options). |                   |             |

</TabItem>
<TabItem value="Traffic-Generic-Id" label="Traffic-Generic-Id">

| Macro        | Description                                                                                                                                            | Default value     | Mandatory   |
|:-------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | INTERFACEID  | Define the interface filter on IDs (OID indexes, e.g.: 1,2,...). If empty, all interfaces will be monitored.  To filter on interface names, see --name |                   |             |

Reviewer: ```suggestion
| INTERFACEID  | Define the interface filter on IDs (OID indexes, e.g.: 1,2,...). If empty, all interfaces will be monitored. To filter on interface names, see --name |                   |             |
```

| CRITICALOUT  | Threshold                                                                                                                                              | 90                |             |
| WARNINGOUT   | Threshold                                                                                                                                              | 80                |             |
| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options).                                                     |                   |             |

</TabItem>
<TabItem value="Traffic-Generic-Name" label="Traffic-Generic-Name">

| Macro         | Description                                                                                                                                            | Default value     | Mandatory   |
|:--------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | INTERFACENAME | Define the interface filter on IDs (OID indexes, e.g.: 1,2,...). If empty, all interfaces will be monitored.  To filter on interface names, see --name |                   |             |

Reviewer: ```suggestion
| INTERFACENAME | Define the interface filter on IDs (OID indexes, e.g.: 1,2,...). If empty, all interfaces will be monitored. To filter on interface names, see --name |                   |             |
```

| WARNINGOUT    | Threshold                                                                                                                                              | 80                |             |
| EXTRAOPTIONS  | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options).                                                     |                   |             |

</TabItem>
<TabItem value="Traffic-Global" label="Traffic-Global">

| Macro          | Description                                                                                                                                                                                                         | Default value     | Mandatory   |
|:---------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | FILTER         | Define the interface filter on IDs (OID indexes, e.g.: 1,2,...). If empty, all interfaces will be monitored.  To filter on interface names, see --name                                                              | .*                |             |

Reviewer: ```suggestion
| FILTER         | Define the interface filter on IDs (OID indexes, e.g.: 1,2,...). If empty, all interfaces will be monitored. To filter on interface names, see --name                                                              | .*                |             |
```
